compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/symbols/ParameterizedTypeSymbol.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.ballerina.compiler.api.symbols;
+
+/**
+ * Represents the parameterized type descriptor.
+ *
+ * @since 2.0.0
+ */
+public interface ParameterizedTypeSymbol extends TypeSymbol {
```

Comment:
I think we should reconsider adding a separate symbol for dependent types since this `BParameterizedType` is an internal type used in the compiler. The spec says the following:

> A return-type-descriptor may be depend on the param-list in the following way. A type-reference occurring in a return-type-descriptor can refer to a parameter name if the type of the parameter is a subtype of typedesc. If p is such a parameter, then a reference to p in the return-type-descriptor denotes the type that is the value of p.

I think it's best to go with how the spec says and use `TypeReferenceTypeSymbol` to model the parameterized type.

Reply:
Updated this as discussed offline in 03a8210 and 6031870, thanks!